web/thesauruses/c/17/strings.json

```diff
@@ -0,0 +1,302 @@
+        "char str[100];",
+        "gets(str);",
+        "printf(\"%s\",strlwr(str);"
+      ],
+      "name": "Convert string to all lowercase"
+    },
+    "convert_to_title_case": {
+      "not-implemented": "true",
+      "name": "Convert string to title case"
+    },
+    "capitalize_string": {
+      "not-implemented": "true",
+      "name": "Capitalize first letter of a string"
+    },
+    "remove_whitespace": {
+      "not-implemented": "true",
+      "comment":"isspace() is available for custom implementation",
+      "name": "Remove all whitespaces from string"
+    },
+    "replace_substring": {
+      "not-implemented": "true",
+      "name": "Replace a substring with another string"
+    },
+    "replace_all_substring": {
+      "not-implemented": "true",
+      "name": "Replace all substring matches with another string"
+    },
+    "split_at_index": {
+      "not-implemented": "true",
+      "name": "Split string into a list of strings at a given index"
+    },
+    "split_at_newlines": {
+      "not-implemented": "true",
+      "name": "Split string into a list of strings at every new line character"
+    },
+    "split_at_substring": {
+      "not-implemented": "true",
+      "name": "Split string by locating all substrings"
+    },
+    "merge_lists_into_string": {
+      "not-implemented": "true",
+      "name": "Merge a list of strings into one string"
+    },
+    "encode_html_entities": {
+      "not-implemented": "true",
+      "name": "Encode HTML entities in a string (ex: ™ to &trade;)"
+    },
+    "decode_html_entities": {
+      "not-implemented": "true",
+      "name": "Decode HTML entitles into characters"
+    },
+    "encode_url_percent": {
+      "not-implemented": "true",
+      "name": "Encode URL percent encoding into string (ex: ' ' to %20)"
+    },
+    "decode_url_percent": {
+      "not-implemented": "true",
+      "name": "Decode URL percent encoding"
+    },
+    "encode_to_base64": {
+      "not-implemented": "true",
+      "name": "Encode string into Base64 format"
+    },
+    "decode_from_base64": {
+      "not-implemented": "true",
+      "name": "Decode string from Base64 format"
+    },
+    "format_string_function": {
+      "not-implemented": "true",
+      "name": "Function to format a string"
+    },
+    "parameter_format_in_order": {
+      "not-implemented": "true",
+      "name": "Parameter used in format function if they're used in order"
+    },
+    "parameter_format_numerical": {
+      "not-implemented": "true",
+      "name": "Parameter used in format function if they're numerically numbered"
+    },
+    "parameter_format_by_name": {
+      "not-implemented": "true",
+      "name": "Paramater used in format function if they're named variables"
+    },
+    "format_as_integer": {
+      "code":[
+        "printf(\"%d\", age); // %i can be also used",
+        "printf(\"%u\", age); // %u for unsigned integer format"
+      ],
+      "name": "Format parameter as an integer"
+    },
```

Comment:
I guess this is a bit vague. `%d` is for signed integers, `%i` for unsigned, and perhaps these should be two different things. I can add this into the meta language file.

but for now, maybe clear it up by saying which type of int `%d/%i` are and the differences.

Reply:
I am a little confused here. I read a few articles on internet regarding different types of format specifiers in C. Some suggest %i as signed some as unsigned and some are neutral. But %d for signed and %u for unsigned are consistent across all.
So, I think only these two should be there to clear confusion. 

what's your say?